pennylane/gradients/hybrid_pulse_gradient.py

```diff
@@ -0,0 +1,564 @@
+            outputted by this transform. Note that this argument doesn't influence the shots
+            used for tape execution, but provides information about the shots.
+        atol (float): Precision parameter used to truncate the Pauli basis coefficients
+            of the effective generators. Coefficients ``x`` satisfying
+            ``qml.math.isclose(x, 0., atol=atol, rtol=0)=True`` are neglected.
+
+    Returns:
+        function or tuple[list[QuantumTape], function]:
+
+        - If the input is a QNode, an object representing the Jacobian (function) of the QNode
+          that can be executed to obtain the Jacobian.
+          The type of the Jacobian returned is either a tensor, a tuple or a
+          nested tuple depending on the nesting structure of the original QNode output.
+
+        - If the input is a tape, a tuple containing a
+          list of generated tapes, together with a post-processing
+          function to be applied to the results of the evaluated tapes
+          in order to obtain the Jacobian.
+
+    **Example**
+
+    Consider the parameterized Hamiltonian
+    :math:`\theta_0 Y^{(0)}+f(\boldsymbol{\theta_1}, t) Y^{(1)} + \theta_2 Z^{(0)}X^{(1)}`
+    with parameters :math:`\theta_0 = \frac{1}{5}`,
+    :math:`\boldsymbol{\theta_1}=\left(\frac{3}{5}, \frac{1}{5}\right)^{T}` and
+    :math:`\frac{2}{5}` as well as a time interval :math:`t=[\frac{1}{10}, \frac{9}{10}]`.
+
+    .. code-block:: python
+
+        jax.config.update("jax_enable_x64", True)
+        H = (
+            qml.pulse.constant * qml.PauliY(0)
+            + jnp.polyval * qml.PauliY(1)
+            + qml.pulse.constant * (qml.PauliZ(0)@qml.PauliX(1))
+        )
+        params = [jnp.array(0.2), jnp.array([0.6, 0.2]), jnp.array(0.4)]
+        t = [0.1, 0.9]
+
+    For simplicity, consider a pulse program consisting of this single pulse and a
+    measurement of the expectation value of :math:`X^{(0)}`.
+
+    .. code-block:: python
+
+        dev = qml.device("default.qubit.jax", wires=2)
+
+        @qml.qnode(dev, interface="jax", diff_method=qml.gradients.hybrid_pulse_grad)
+        def circuit(params):
+            op = qml.evolve(H)(params, t)
+            return qml.expval(qml.PauliX(0))
+
+    We registered the ``QNode`` to be differentiated with the ``hybrid_pulse_grad`` method.
+    This allows us to simply differentiate it with ``jax.jacobian``, for example, internally
+    making use of the hybrid pulse parameter-shift method.
+
+    >>> jax.jacobian(circuit)(params)
+    [Array(1.4189798, dtype=float32, weak_type=True),
+     Array([0.00164918, 0.00284802], dtype=float32),
+     Array(-0.09984542, dtype=float32, weak_type=True)]
+
+    Alternatively, we may apply the transform to the tape of the pulse program, obtaining
+    the tapes with inserted ``PauliRot`` gates together with the post-processing function:
+
+    >>> tapes, fun = qml.gradients.hybrid_pulse_grad(circuit.tape, argnums=[0, 1, 2])
```

Comment:
I was thinking about this as well..
I think it is nicer to have the `jax.jacobian` example first, because this is definitely the way users should use the function. Would it help to explicitly call `circuit.construct`, or to manually create an entirely new tape?

Reply:
Either of those would work, yep! Dont think I have a personal preference :)